### ISSUE TITLE
replace the H.A.D. links with something better

### ISSUE THREAD

**Issue by jfriedel** [OpenGrok/OpenGrok#1366]
Remove the H.A.D links from the list and substitute them with a window of links. The window should show up when one hovers over the respective file in the list. The window should have active links so that one can click on them to open the respective item (e.g. open a full history in a new tab). Some key combination could be used (say holding shift) to keep the window open while moving with the mouse pointer from the file to the respective window, so that a) the window does not close b) the window contents does not change when a user accidentally hovers over some other file in the list.

![win_link](https://cloud.githubusercontent.com/assets/5879789/22464756/c016c9a4-e7b8-11e6-9e95-0a100a6ac1f7.jpg)

**Comment by kosi2801** [comment]
I personally would vote against this specific proposal.
On the one side having the popup appear automatically over the results
lowers the readability when scrolling over the results/snippets and on the
other hand impairs usability quite a bit if one is required to hold a key
when trying to click one of the offered links.

Furthermore the proposed popup seems to concentrate on Git as changesets
are probably not available for other VSC.

I'm not against changing the behavior if it doesn't impact readability and
usability. Probably such a popup could be displayed when hovering/clicking
a certain icon instead of the "H A D" links and only disappear if another
icon or a space outside of the popup is clicked. This would also keep the
usability on pointerless devices (e.g. mobile phones, touch displays, ...)
intact.
But for my current usage patterns of OpenGrok the "H A D" links leave
little possibility to improve without having to introduce additional clicks.

**Comment by jfriedel** [comment]
Yep; probably a better solution should be implemented -- however, the basic idea of lowering down the significance of the "H A D" field should be implemented.

Changeset (or a like, unique identifiers) are used by Mercurial as well.  In the proposal above, one can read the changeset_A as the latest change, changeset_B as the last but one change in the given file, etc. This should be understandable for both - the ancient file based VCSs and the changeset based VCSs.

**Comment by vladak** [comment]
Although I've been user of these links, their appearance is weird in all the views. Putting up new popup window feels wrong, though. I'd welcome some better UI elements.

**Comment by jfriedel** [comment]
NAGI.

**Comment by vladak** [comment]
I still think there needs to be some more thought and debate about how to make this better.
